lang/en/typeshed/stdlib/time.pyi

```diff
@@ -56,6 +56,16 @@ def ticks_us() -> int:
     """
     ...
 
+def ticks_cpu() -> int:
+    """
+    Similar to ticks_ms and ticks_us, but with higher resolution in CPU cycles.
+
+    Example: ``time.ticks_cpu()``
+
+    :return: The counter value in CPU cycles.
+    """
+    ...
+
```

Comment:
Hi @microbit-carlos, this is the best guess we made for `ticks_cpu` because we couldn't find micropython docs on it. Is this right and are there any more specific information we can add about this? Thanks!

Reply:
Hi Grace,

I wasn't aware that the `ticks_cpu` function was enabled in the V2 MicroPython port, do you know when it was added? Has it been added only in the latest release?

There is some upstream documentation here: https://docs.micropython.org/en/latest/library/time.html#time.ticks_cpu

But it would be good to understand the exact implementation for micro:bit and to have that in the official docs as well. I'll have a look.

Reply:
Upgraded here: https://github.com/microbit-foundation/micropython-microbit-v2/commit/fdaf840e86ed4721123b8b1fddb4830e387532b2

That diff contains the implementation that uses CYCCNT.

Reply:
For the moment we always return zero in the sim for ticks_cpu but we could return the same as ticks_us or a multiple. ticks_us is already a 1000 * multiple of ticks_ms which is our real resolution in the sim.

Reply:
Looks like it was added with the update to MicroPython 1.22.0 in January: https://github.com/microbit-foundation/micropython-microbit-v2/pull/166
It uses low level Arm register to count cycles, so this should be correct (of course, there will be a bit of overhead by calling and returning from a Python function but that's the same for every other time.ticks_xx function):
https://github.com/microbit-foundation/micropython-microbit-v2/blob/8d9067d91bcfe316cd3aea5af1a873a37ca23d13/src/codal_port/mphalport.h#L48-L55

The thing to keep in mind is that this is register is 32 bits and we run at 64 MHz, so the register overflows every minute ish ( 2^32 / 64_000_000 = 67 seconds)

Reply:
Thanks @microbit-carlos !

Reply:
Correction, it's likely 30-bits due to small int size in MicroPython.